File: ruby/raindrops/_iterations/1/raindrops.rb
class Raindrops
  def self.convert(number)
    return_value = ""
    if number % 3 == 0
      return_value = return_value + "Pling"
    end
    if number % 5 == 0
      return_value = return_value + "Plang"
    end
    if number % 7 == 0
      return_value = return_value + "Plong"
    end
    return_value.empty? ? "#{number}" : return_value
  end  
end

File: typescript/raindrops/_iterations/2/raindrops.ts
export function convert(input: number): string {
  let result = "";
  if (input % 3 === 0) result += "Pling";
  if (input % 5 === 0) result += "Plang";
  if (input % 7 === 0) result += "Plong";
  return result ? result : String(input);
}
